components/automate-ui/src/app/pages/project/rules/project-rules.component.spec.ts

```diff
@@ -254,5 +255,17 @@ describe('ProjectRulesComponent', () => {
       component.addCondition();
       expect(component.showDelete()).toBeTruthy();
     });
+
+    fit('should show "Save Rule" Button by default', () => {
+      const componentDebug: DebugElement = fixture.debugElement.nativeElement;
+      const confirmButton = componentDebug.querySelector('#right-buttons button:first-child');
+      expect(confirmButton).toContain('Save Rule');
+    });
+
```

Comment:
@msorens - I can't seem to find the correct way to query the DOM in my tests.  I've searched around and have mainly been following Angular's guide lines [on using debugElement](https://angular.io/guide/testing#debugelement), but I can seem to query inside the root element.  Any tips?

Reply:
Ah, I suspect the wall you are hitting is Stencil; good chance that Stencil components are opaque to such probing. But you actually have what you need without penetrating that component. Try something like this:
```
      const element = fixture.nativeElement;
      expect(element.querySelector('chef-page')
        .getAttribute('confirm-btn-text')).toEqual('Save Rule');
```

Reply:
Thank you @msorens - That works!  I'll look deeper into testing stencil components 👍